automata/fa/dfa.py

```diff
@@ -591,12 +707,16 @@ def isfinite(self):
         ))
 
         important_nodes = accessible_nodes.intersection(coaccessible_nodes)
-
+        subgraph = G.subgraph(important_nodes)
         try:
-            nx.find_cycle(G.subgraph(important_nodes))
-            return False
-        except nx.exception.NetworkXNoCycle:
-            return True
+            return nx.dag_longest_path_length(subgraph)
+        except nx.exception.NetworkXUnfeasible:
+            return float('inf')
+
+    @staticmethod
+    def _stringify_states_unsorted(states):
```

Comment:
I think this is here by accident? We removed this function a little while ago.

Reply:
Merge accident

Reply:
@Tagl Oh, I see. Sorry about that—that was an artifact I must've introduced during the rebase.